### PR TITLE
chore(TS): add typescript definitions

### PR DESCRIPTION
**Pull request by TylorS**

### CODE REVIEW

File: type-definitions/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  catch<U>(onRejected?: (error: any) => U | Thenable<U>): Promise<U>;
+}
+
+declare interface DisposeFn {
+  (): void | Promise<any>;
+}
+
+export default function create<A>(f: (add: (a: A) => any,
+                                      end: (x: any) => any,
```
Comment (briancavalier) [inline type-definitions/index.d.ts]: Should we make this match the ones in most?  i.e. `end: (x: ?A) => any`?

Comment (TylorS) [inline type-definitions/index.d.ts]: Yes!

**Comment by TylorS** [comment]
Fixed!

File: type-definitions/index.d.ts [hunk 2]
```diff
@@ -20,5 +20,5 @@ declare interface DisposeFn {
 }
 
 export default function create<A>(f: (add: (a: A) => any,
-                                      end: (x: any) => any,
+                                      end: (x?: any) => any,
```
Comment (briancavalier) [inline type-definitions/index.d.ts]: Needs to be of type `A`, tho.  These are basically "functionified" versions of the [Sink methods](https://github.com/cujojs/most/blob/master/type-definitions/most.d.ts#L24), so I think we can make the return type `void`:

``` js
export default function create<A>(f: (add: (a: A) => void,
                                      end: (a?: A) => void,
                                      error: (e: Error) => void) => void | DisposeFn ): Stream<A>;
```

Comment (TylorS) [inline type-definitions/index.d.ts]: Yeah okay, I'll make the changes

**Comment by TylorS** [comment]
Fixed!

**Comment by briancavalier** [comment]
Looks good, thanks!

**Pull request merged**